tests/conftest.py

```diff
@@ -37,10 +37,27 @@ def pytest_runtest_setup(item):
         mock.patch.object(symbol_database, "Default", return_value=sym_db),
     ]
     if descriptor_pool._USE_C_DESCRIPTORS:
-        from google.protobuf.pyext import _message
+        has_upb = False
+        try:
+            from google._upb import _message
+
+            has_upb = True
+            assert _message
+        except ImportError:
+            _message = None
+        if not _message:
+            try:
+                from google.protobuf.pyext import _message
+            except ImportError:
+                _message = None
```

Comment:
Consider having a utility helper method or something similar for htis logic, as it seems it is being reused in multiple places